napi_error_handling.md

```diff
@@ -0,0 +1,115 @@
+# Error handling
+
+N-API uses both return values and exceptions for error handling.  The sections
```

Comment:
Maybe clarify up front that JavaScript exceptions are used, but C++ exceptions are not.